core_lib_generator/config_collectors/cache.py

```diff
@@ -7,40 +7,76 @@ def generate_cache_template() -> dict:
     cache_name = input_str('Enter name for your cache')
     cache_type = input_enum(CacheTypes, 'From the following list, what cache will you use?', CacheTypes.Memory.value)
 
-    if cache_type == CacheTypes.Memcached.value:
-        memcache_port = input_int('Enter your memcached server port no.', 11211)
-        memcache_host = input_str('Enter your memcached server host', 'localhost')
-        print(f'Cache type {CacheTypes(cache_type).name} on {memcache_host}:{memcache_port}')
+    if cache_type == CacheTypes.Memory.value:
+        print(f'Cache type {CacheTypes(cache_type).name}')
+        return {
+            'config': {
+                cache_name: {
+                    'type': CacheTypes(cache_type).name.lower(),
+                }
+            }
+        }
+    elif cache_type == CacheTypes.Empty.value:
+        print('No cache set.')
+    else:
+        cache_type_name = CacheTypes(cache_type).name
+        cache_port = input_int(f'Enter your {cache_type_name} server port no.', default_cache_ports[cache_type_name])
+        cache_host = input_str(f'Enter your {cache_type_name} server host', 'localhost')
+        cache_protocol = None
+        if cache_type_name == CacheTypes.Redis.name:

```

Comment:
what is the reason that here you are detecting the redis type using the `cache_type_name`
and in all other places using the `cache_type`

again, 
please make all logic the same.